server/resp/src/main/java/org/infinispan/server/resp/commands/generic/EXISTS.java

```diff
@@ -0,0 +1,44 @@
+package org.infinispan.server.resp.commands.generic;
+
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.util.concurrent.AggregateCompletionStage;
+import org.infinispan.util.concurrent.CompletionStages;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * EXISTS Resp Command
+ *
+ * @link https://redis.io/commands/exists/
+ * @since 15.0
+ */
+public class EXISTS extends RespCommand implements Resp3Command {
+   public EXISTS() {
+      super(-2, 1, -1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+         ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      AggregateCompletionStage<Void> acs = CompletionStages.aggregateCompletionStage();
+      AtomicLong presentCount = new AtomicLong(arguments.size());
+      for (byte[] bs : arguments) {
+         acs.dependsOn(handler.cache()
+               .computeIfAbsentAsync(bs, (v) -> {
```

Comment:
This won't quite work. I can think of an edge case where if an entry is expired this will cause a REMOVE event when it should really be an EXPIRED event. I would say to use `Cache.containsKeyAsync` but we don't have that method. You should be able to use `AdvancedCache.touch(key, false);` instead though.

Reply:
ok I've changed to "touch", also implemented the method in the cache